core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -491,6 +495,19 @@ abstract class AbstractFetcherThread(name: String,
     } finally partitionMapLock.unlock()
   }
 
+  def addTopicIdsToThread(partitions: Set[TopicPartition], topicIds: String => Option[Uuid]) = {
```

Comment:
nit: Should we prefix this method with `maybe` to indicate that it would set the topic id only if there is a state for the topic partition?